lib/spi/types.ts

```diff
@@ -80,6 +80,7 @@ export interface ApplicationRepository extends GitRepositoryReference {
 export class ResourceContext {
 
     private readonly resources: Map<string, cdk.IResource> = new Map();
+    removalPolicy: cdk.RemovalPolicy | undefined;
```

Comment:
Why do we need this here? Where is it used?

Reply:
My apologies, I reversed all the unnecessary changes in the code due to the auto-formatting.

Reply:
addressed